src/client/al-api-client.ts

```diff
@@ -569,14 +602,27 @@ export class AlApiClient implements AlValidationSchemaProvider
 
   /**
    * Resolves accumulated endpoints data for the given account.
+   *
+   * Update Feb 2021
+   * ---------------
+   * This has been overhauled to deal with services whose endpoints now must be determined for the current context residency (selected datacenter location in UI).
+   * The reason for this is to cater for scenarios where a parent account manages children that are located across different geographical locations to one another and therefore
+   * any data retrieval for views in the UI where child account roll-ups exist must be fetched from the appropriate location.
+   * The previous implementation ALWAYS calculated service endpoints for the default location of the primary account for the logged in user and so any roll ups for views for child accounts never worked eva!!!
+   *
    */
-  public async getServiceEndpoints( accountId:string, requestList?:string[] ):Promise<AlEndpointsServiceCollection> {
+  public async getServiceEndpoints( accountId:string, requestList:string[], resolveByResidency = false ):Promise<AlEndpointsServiceCollection | AlEndpointsResidencyServiceCollection> {
     const environment = AlLocatorService.getCurrentEnvironment();
-    const cacheKey = `/endpoints/${environment}/${accountId}`;
+    const context = AlLocatorService.getContext();
+    const cacheKey = resolveByResidency ? `/endpointsByResidency/${environment}/${accountId}` : `/endpoints/${environment}/${accountId}`;
+    let existingEndpoints: AlEndpointsServiceCollection | AlEndpointsResidencyServiceCollection;
+    let translated: AlEndpointsServiceCollection | AlEndpointsResidencyServiceCollection;
+
     if ( ! requestList ) {
       requestList = AlApiClient.defaultServiceList;
     }
-    let existingEndpoints = this.getCachedValue<AlEndpointsServiceCollection>( cacheKey );
+    console.log('getServiceEndpoints - getting endpoints from - ' + cacheKey);
```

Comment:
All that reworking of `this.log` and you just plop this console log down here?  Ain't right, man.